File: Programmers/practice14.py
def solution(n):
    answer = (int(n ** 0.5) + 1) ** 2 if int(n ** 0.5) ** 2 == n else -1
    return answer

n = 121
# answer = 144
print(solution(n))

n = 3
# answer = -1
print(solution(n))

File: Programmers/practice14.js
function solution(num) {
    var answer = 0;

    for(let i = 0; i < 500; i++){
        if(num % 2 === 0){
            num = Math.floor(num / 2);
        } else if(num === 1){
            return answer;
        } else {
            num = num * 3 + 1;
        }
        answer++;
    }

    return -1;
}